core/parentchain/indirect-calls-executor/src/event_filter.rs

```diff
@@ -68,6 +97,27 @@ impl FilterEvents for Events<H256> {
 			})
 			.collect())
 	}
+
+	fn get_transfer_events(&self) -> Result<Vec<BalanceTransfer>> {
+		Ok(self
+			.iter()
+			.filter_map(|ev| {
+				log::info!("GOT BEFORE AND_THEN");
+				ev.and_then(|ev| match ev.as_event::<BalanceTransfer>()? {
+					Some(e) => {
+						log::info!("GOT IN SOME MATCH");
+						Ok(Some(e))
+					},
+					None => {
+						log::info!("GOT IN NONE MATCH");
+						Ok(None)
+					},
+				})
+				.ok()
+				.flatten()
+			})
+			.collect())
```

Comment:
This does not seem natural, but I guess it will be much simpler without the debug logs anyhow. Something like the following should work:

```rust
 self.iter()
   .flatten() // flatten filters out the nones
   .filter_map(|ev| ev.as_event::<BalanceTransfer>) // I think we could just log the result instead of within the closure.
   .collect()
```

What do you think?

Reply:
081a456b5ad8f96d2e10a8bbf40dcbacc284e682